proposed/2022/pack-referenced-projects.md

```diff
@@ -0,0 +1,71 @@
+# Pack referenced projects
+
+- Author Name [maxkoshevoi](https://github.com/maxkoshevoi)
+- Start Date (2022-03-28)
+- GitHub Issue [3891](https://github.com/NuGet/Home/issues/3891)
+- GitHub PR (-)
+
+## Summary
+
+Allow referenced projects' artifacts to be packed inside the NuGet.
+
+## Motivation
+
+When SDK-style project is being packed as a NuGet, all projects that it references are considered as NuGet themselves that this NuGet depends on.
+This is not always a desired behavior, sometimes user needs artifacts from referenced project(s) to be packed into the NuGet.
+Currently there's no easy way to override this behavior except for ditching `csproj`, and going back to `nuspec`. In many cases that involves piping lots of information from the `MSBuild` context into the `nuspec` evaluation via `key=value` parameters.  
+Chaining together dozens of arguments to achieve parity with metadata usually automagically introduced by the SDK becomes tedious, which eventually weakens the otherwise rich set of information attached to a modern NuGet package.
+
+## Explanation
+
+### Functional explanation
+
+Proposed solution is to add new `Pack` and `PackagePath` properties to `ProjectReference` (`<ProjectReference Include="..." Pack="True" PackagePath="tools" />`) and `PackageReference`. 
+If `Pack` is `False` or not present - consider the project (or package) as a NuGet's dependency, if it's `True` - pack its artifacts in.
+If `PackagePath` is not present - pack artifacts to the default location within the package. If it's present - pack them to the specified path.
+
+For `PackageReference`s it's useful in case referenced package is from private feed, but NuGet that's being packed will be public.
```

Comment:
nit: This needs to go into the motivation.

Reply:
Additionally an aside, packing assemblies from packages published on private feeds only can be lead to a lot of issues.

If a package is in a private feed, it must have been done so for a reason. Redistributing assemblies for packages you don't own might lead into some licensing violations.

There's no guarantee that this package currently on a private feed (maybe because it's in previews) does not make it to NuGet.org. Redistributing the `same assembly in multiple packages` is something that's known to cause a lot of problems for the consumers.
The build and runtime work best when only 1 reference of the same assembly is used in an application. NuGet ensures that you cannot have multiple versions of the package in the project, but that guarantee cannot be transferred to the assemblies themselves. If an assembly is in multiple packages, it's very possible that someone ends up referencing both. The more dependency issues can be resolved at restore time the  better.

The above concerns can be generalized to `Never include the same assembly in multiple library packages`.
Note the focus on library packages. In some scenarios, packages contain tools, and those often need the entry assembly + some of it's dependencies.